druntime/src/rt/aaA.d

```diff
@@ -210,182 +215,58 @@ Bucket[] allocBuckets(size_t dim) @trusted pure nothrow
 
 private void* allocEntry(scope const Impl* aa, scope const void* pkey)
 {
-    import rt.lifetime : _d_newitemU;
     import core.stdc.string : memcpy, memset;
 
-    immutable akeysz = aa.valoff;
-    void* res = void;
-    if (aa.entryTI)
-        res = _d_newitemU(aa.entryTI);
-    else
+    immutable akeysz = aa.keysz;
+    immutable allocsz = aa.valoff + aa.valsz;
+    auto flags = (aa.flags & Impl.Flags.hasPointers) ? 0 : GC.BlkAttr.NO_SCAN;
+    if (aa.flags & Impl.Flags.hasDtors)
     {
-        auto flags = (aa.flags & Impl.Flags.hasPointers) ? 0 : GC.BlkAttr.NO_SCAN;
-        res = GC.malloc(akeysz + aa.valsz, flags);
+        assert(aa.entryTI !is null);
+        flags |= GC.BlkAttr.FINALIZE;
     }
+    auto res = GC.malloc(allocsz, flags, aa.entryTI);
```

Comment:
AFAICT this is now the wrong type info passed to malloc, use null for conservative scanning.

Reply:
The `entryTI` is null, or  the `Entry` struct TI (for [statically allocated AAs](https://github.com/dlang/dmd/blob/c73dca185a5b122ad84a18560661f72dee5de3c8/druntime/src/core/internal/newaa.d#L78)), or the AA itself, which returns [`rtinfoHasPointers`](https://github.com/dlang/dmd/blob/c73dca185a5b122ad84a18560661f72dee5de3c8/druntime/src/object.d#L758).